Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -689,21 +689,20 @@ public final class ManifestLoader: ManifestLoaderProtocol {
         let moduleCachePath = (ProcessEnv.vars["SWIFTPM_MODULECACHE_OVERRIDE"] ?? ProcessEnv.vars["SWIFTPM_TESTS_MODULECACHE"]).flatMap{ AbsolutePath.init($0) }
 
         var cmd: [String] = []
-        cmd += [self.toolchain.swiftCompiler.pathString]
+        cmd += [self.toolchain.swiftCompilerPath.pathString]
         cmd += verbosity.ccArgs
 
         let macOSPackageDescriptionPath: AbsolutePath
-        // If we got the binDir that means we could be developing SwiftPM in Xcode
+        // if runtimePath is set to "PackageFrameworks" that means we could be developing SwiftPM in Xcode
         // which produces a framework for dynamic package products.
-        let packageFrameworkPath = runtimePath.appending(component: "PackageFrameworks")
-        if self.toolchain.binDir != nil, localFileSystem.exists(packageFrameworkPath)  {
+        if runtimePath.basename == "PackageFrameworks" {
```

Comment:
@abertelrud I dont love this, but its not much worst than before. any better ideas?

Reply:
Probably for a separate PR, but in general I think it would be good to have platform peculiarities that are only needed at debug time be handled through the array of custom compiler arguments.  In other words, Xcode (or whatever IDE) would determine what special sauce it needs at debug time and then pass that to libSwiftPM.  I realize that's pretty vague, but if we can find a way to do that (agian, in a separate PR) that would clean this up.

Reply:
agreed